app/src/main/java/com/hedvig/app/feature/offer/ui/OfferAdapter.kt

```diff
@@ -418,4 +498,27 @@ class OfferAdapter(
             }
         }
     }
+
+    companion object {
+        const val INSURELY_CARD = 1
+        const val CURRENT_INSURANCES_HEADER = 2
+        const val INSURELY_DIVIDER = 3
+
+        class OfferDiffUtilCallback : DiffUtil.ItemCallback<OfferModel>() {
+            override fun areItemsTheSame(oldItem: OfferModel, newItem: OfferModel): Boolean = when {
+                oldItem is OfferModel.InsurelyCard && newItem is OfferModel.InsurelyCard -> {
+                    oldItem.id == newItem.id
+                }
+                oldItem is OfferModel.CurrentInsurancesHeader && newItem is OfferModel.CurrentInsurancesHeader -> {
+                    // Should only display 1 CurrentInsurancesHeader ever
+                    true
+                }
+                else -> {
+                    oldItem == newItem
+                }
+            }
+
+            override fun areContentsTheSame(oldItem: OfferModel, newItem: OfferModel) = oldItem == newItem
```

Comment:
Can we check if contents are same here?

Reply:
Since every single OfferModel subclass is a data class they should be handled properly with the built-in == implementation, so this should work as intended. Just like the GenericDiffUtilItemCallback does